src/SuperSocket.Connection/PipeConnectionBase.cs

```diff
@@ -412,5 +423,16 @@ protected void OnError(string message, Exception e = null)
             else
                 Logger?.LogError(message);
         }
+
+        protected void SupplyEnd()
```

Comment:
These methods are not required by other derived classes, so probably you can leave them in KestrelPopeConnection.